src/test/java/teammates/test/cases/datatransfer/CourseRosterTest.java

```diff
@@ -104,15 +105,96 @@ public void testGetEmailToNameTableFromRoster() {
+        assertEquals("John", info[CourseRoster.INFO_TEAM_NAME]);
+        assertEquals("John's Section", info[CourseRoster.INFO_SECTION_NAME]);
+    }
+
+    @Test
+    public void testGetInfoForIdentifier_instructorCase_shouldShowCorrectInfo() {
+        CourseRoster roster = new CourseRoster(
+                createStudentList(
+                        "s1", "s1@gmail.com",
+                        "s2", "s2@gmail.com",
+                        "s3", "s3@gmail.com"),
+                createInstructorList(
+                        "John", "john@email.com",
+                        "Jean", "ins2@email.com"));
+        String[] info = roster.getInfoForIdentifier("john@email.com");
+        assertEquals("John", info[CourseRoster.INFO_NAME]);
+        assertEquals(Const.USER_TEAM_FOR_INSTRUCTOR, info[CourseRoster.INFO_TEAM_NAME]);
+        assertEquals(Const.DEFAULT_SECTION, info[CourseRoster.INFO_SECTION_NAME]);
+    }
+
+    @Test
+    public void testGetInfoForIdentifier_teamCase_shouldShowCorrectInfo() {
+        CourseRoster roster = new CourseRoster(
+                createStudentList(
+                        "s1", "s1@gmail.com",
+                        "s2", "s2@gmail.com",
+                        "s3", "s3@gmail.com"),
+                createInstructorList(
+                        "John", "john@email.com",
+                        "Jean", "ins2@email.com"));
+        String[] info = roster.getInfoForIdentifier("s1");
+        assertEquals("s1", info[CourseRoster.INFO_NAME]);
+        assertEquals("s1", info[CourseRoster.INFO_TEAM_NAME]);
+        assertEquals("s1's Section", info[CourseRoster.INFO_SECTION_NAME]);
+    }
+
+    @Test
+    public void testGetInfoForIdentifier_unknownCase_shouldShowCorrectInfo() {
+        CourseRoster roster = new CourseRoster(
+                createStudentList(
+                        "s1", "s1@gmail.com",
+                        "s2", "s2@gmail.com",
+                        "s3", "s3@gmail.com"),
+                createInstructorList(
+                        "John", "john@email.com",
+                        "Jean", "ins2@email.com"));
+        String[] info = roster.getInfoForIdentifier("random");
+        assertEquals(Const.USER_NOBODY_TEXT, info[CourseRoster.INFO_NAME]);
+        assertEquals(Const.USER_NOBODY_TEXT, info[CourseRoster.INFO_TEAM_NAME]);
+        assertEquals(Const.DEFAULT_SECTION, info[CourseRoster.INFO_SECTION_NAME]);
+    }
+
     private List<StudentAttributes> createStudentList(String... studentData) {
         List<StudentAttributes> students = new ArrayList<>();
         for (int i = 0; i < studentData.length; i += 2) {
             StudentAttributes student = StudentAttributes
-                    .builder("", "")
-                    .withName("")
+                    .builder("", studentData[i + 1])
+                    .withName(studentData[i])
+                    .withTeamName(studentData[i])
+                    .withSectionName(studentData[i] + "'s Section")
```

Comment:
Although it is just a test class, I think it is still good to name each element in the pair within the for loop for better understanding.
Like `studentTeam = studentData[i]`, `studentEmail = studentData[i + 1]`.

Reply:
The names are implicitly specified by the builder pattern?

Reply:
There's no indication what `studentData[i + 1]` is when instantiating the builder.